### PR TITLE
Fix and enhancement for generate outputs command

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47046
fixes: https://github.com/demisto/etc/issues/47236
fixes: https://github.com/demisto/etc/issues/47246
fixes: https://github.com/demisto/etc/issues/47248

## Description

- Fixing the command to work
- Fixing `-o` argument in genereate outputs example flow
- Providing multiple examples of same command
- Adding outputs to existing YML without overriding it (keeping the description, for example)

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py [hunk 1]
```diff
@@ -38,44 +38,72 @@ def generate_example_dict(examples_file: Optional[str], insecure=False):
...
+    if command_name not in command_names:
+        raise Exception(f'The {command_name} command is missing from the integration YML.')
+    command_index = command_names.index(command_name)
+    command = commands[command_index]
+
+    outputs: List[Dict[str, str]] = command.get('outputs') or []
+    old_descriptions = _output_path_to_description(outputs)
+    new_descriptions = _output_path_to_description(output_with_contexts)
+
+    old_output_paths: Set = set(filter(None, map(lambda val: val.get('contextPath'), command.get('outputs', []))))
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py:60]: Do not do set from filter. Can't you use it as the iterator it wants to be?

File: demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py [hunk 2]
```diff
@@ -38,44 +38,72 @@ def generate_example_dict(examples_file: Optional[str], insecure=False):
...
     return yml_data
 
 
+def _output_path_to_description(command_outputs: List[Dict[str, str]]) -> Dict[str, str]:
+    """ creates a mapping of contextPath -> description, if the description is not null. """
+    descriptions = {}
+    for output in command_outputs:
+        description = output.get('description')
+        context_path = output.get('contextPath')
+        if context_path and description:
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py:83]: is it fine when it does not exists?

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py [hunk 3]
```diff
@@ -106,6 +93,8 @@ def test_insert_outputs(mocker):
         if command.get('name') == command_name:
             assert command['outputs'] == FAKE_OUTPUT_CONTEXTS
             break
+    else:
+        assert False, f'No command {command_name} in yml_data'
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py:97]:
```suggestion
        raise AssertionError(f'No command {command_name} in yml_data)'
```

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py [hunk 4]
```diff
@@ -148,3 +141,5 @@ def test_generate_integration_context(mocker, tmpdir):
         if command.get('name') == command_name:
             assert command['outputs'] == FAKE_OUTPUT_CONTEXTS
             break
+    else:
+        assert False, 'command is not found in yml_data'
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py:145]:
```suggestion
        raise AssertionError('command is not found in yml_data')
```

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py [hunk 5]
```diff
@@ -136,10 +129,10 @@ def test_generate_integration_context(mocker, tmpdir):
         if command.get('name') == command_name:
             command['outputs'] = ''
             break
+    else:
+        assert False, 'command is not found in yml_data'
 
-    generate_integration_context.generate_integration_context(filename,
-                                                              FAKE_EXAMPLES_FILE,
-                                                              insecure=True,
+    generate_integration_context.generate_integration_context(filename, FAKE_EXAMPLES_FILE, insecure=True,
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py:135]: do you really need to pass insecure in a test?

**Review by jochman — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py (hunk 2 above)
Comment (ilaner) [reply demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py:83]: Every output entry has both 'description' and 'contextPath', but the description could be empty.

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py (hunk 5 above)
Comment (ilaner) [reply demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py]: not really, deleted

File: demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py (hunk 1 above)
Comment (ilaner) [reply demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py]: changed to set comprehension, think it's more readable this way.

File: demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py [hunk 6]
```diff
@@ -38,44 +38,70 @@ def generate_example_dict(examples_file: Optional[str], insecure=False):
...
     Args:
+        output_path: Output path
         input_path: path to the yaml integration.
         examples: path to the command examples.
         insecure: should use insecure.
         verbose: verbose (debug mode).
     """
-
+    if not output_path:
+        print_warning('output path wasn\'t provided, saving in input path')
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/generate_integration_context.py:103]: lets not do warning. lets just print "saving output to <path>" or something like that.

File: demisto_sdk/commands/generate_outputs/generate_context/tests/test_data/fake_outputs1.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/test_data/fake_outputs1.json:1]: please don't use files data. Just drop it as a constant in your test.

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py [hunk 8]
```diff
@@ -136,15 +197,16 @@ def test_generate_integration_context(mocker, tmpdir):
...
+    generate_integration_context.generate_integration_context(filename, FAKE_EXAMPLES_FILE, verbose=False)
 
     # Check we have new data
     yml_data = get_yaml(filename)
     for command in yml_data['script']['commands']:
         if command.get('name') == command_name:
             assert command['outputs'] == FAKE_OUTPUT_CONTEXTS
             break
+    else:
+        raise AssertionError('command is not found in yml_data')
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py:212]: Specify the command that you wanted to find

File: demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py [hunk 9]
```diff
@@ -136,15 +197,16 @@ def test_generate_integration_context(mocker, tmpdir):
         if command.get('name') == command_name:
             command['outputs'] = ''
             break
+    else:
+        AssertionError('command is not found in yml_data')
```
Comment (jochman) [inline demisto_sdk/commands/generate_outputs/generate_context/tests/generate_integration_context_test.py:201]: Specify the command that you wanted to find

**Review by jochman — APPROVED** [review]
🚀

**Pull request merged**